src/agent0/core/hyperdrive/policies/simple_lp.py

```diff
@@ -0,0 +1,172 @@
+
+    def action(
+        self, interface: HyperdriveReadInterface, wallet: HyperdriveWallet
+    ) -> tuple[list[Trade[HyperdriveMarketAction]], bool]:
+        """Specify actions.
+
+        Arguments
+        ---------
+        interface: HyperdriveReadInterface
+            Interface for the market on which this agent will be executing trades (MarketActions).
+        wallet: HyperdriveWallet
+            The agent's wallet.
+
+        Returns
+        -------
+        tuple[list[MarketAction], bool]
+            A tuple where the first element is a list of actions,
+            and the second element defines if the agent is done trading.
+        """
+        # Get the current state of the pool & the bot's position
+        current_block = interface.get_current_block()
+        pool_state = interface.get_hyperdrive_state(current_block)
+        current_holding = wallet.lp_tokens * pool_state.pool_info.lp_share_price
+
+        # Need to be in the game to play it
+        if self.policy_config.minimum_liquidity_tokens > current_holding:
+            trade_amount = self.policy_config.minimum_liquidity_tokens - current_holding
+            return [add_liquidity_trade(trade_amount)], False
+
+        # Get current PNL
+        pnl = current_holding - self.total_base_spent
+        self.pnl_history.append((FixedPoint(interface.get_block_timestamp(current_block)), pnl))
+        # Prune history
+        if FixedPoint(len(self.pnl_history)) > self.policy_config.lookback_length:
+            self.pnl_history = self.pnl_history[-self.policy_config.lookback_length :]
+        # PNL slope is dy/dx, where y is PNL and x is time.
+        # We compute an average of splines instead of a global average
+        # to reduce emphasis on the first and last time-points.
+        avg_pnl_slope = 0
+        if len(self.pnl_history) > 2:
+            for i in range(len(self.pnl_history) - 1):
+                dx = self.pnl_history[i + 1][0] - self.pnl_history[i][0]
+                dy = self.pnl_history[i + 1][1] - self.pnl_history[i][1]
+                avg_pnl_slope += dy / dx
+            avg_pnl_slope /= len(self.pnl_history) - 1
+
+        # Make trades based on pnl_target and avg_pnl_slope
+        action_list = []
+
+        # I'm doing great, keep putting money in
+        if pnl > self.policy_config.pnl_target and avg_pnl_slope > 0:
+            # only add money if you can afford it!
+            if wallet.balance.amount >= self.policy_config.delta_liquidity:
+                action_list.append(add_liquidity_trade(self.policy_config.delta_liquidity))
+
+        # I'm doing bad, time to pull some money out
+        elif pnl < self.policy_config.pnl_target and avg_pnl_slope < 0:
+            # delta_liquidity is in base; convert it to tokens
+            delta_tokens = self.policy_config.delta_liquidity / pool_state.pool_info.lp_share_price
+            # we do not want to pull out so many tokens that we are below the minimum
```

Comment:
to keep it simple, i think you can take out this check. since you actually want the minimum to be triggered, to act as a "reset" to make the bot keep trying. otherwise it will just sit at just outside the threshold and never trade again.

Reply:
It has a minimum liquidity amount, so it'll always have some stake in the game. We want the liquidity in the pool to go down when the LP is unhappy so that traders will also be unhappy due to slippage. So I think we need this to stay in so that the behavior is two-sided.

Reply:
oh right, it doesn't need to "add liquidity" for a chance to improve. it just needs to "keep something open" which this does!